packages/client/src/index.ts

```diff
@@ -17,6 +17,75 @@
  * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
-export * as analyzer from './changeAnalysis';
-export * as functions from './processing';
-export { restClient } from './rest';
+// Exporting zod Schemas and types for the basic dictionary components and data types
+export {
+	DataRecord,
+	DataRecordValue,
+	UnprocessedDataRecord,
+	ArrayDataValue,
+	SingleDataValue,
+	Dictionary,
+	Schema,
+	SchemaField,
+	Result,
+} from '@overture-stack/lectern-dictionary';
+import * as _validation from '@overture-stack/lectern-validation';
+
+// Exporting all result types used in the parsing and validation functions
+export type {
+	TestResult,
+	TestResultValid,
+	TestResultInvalid,
+	SchemaRecordError,
+	DictionaryValidationError,
+	FieldValidationError,
+	ParseDictionaryData,
+	ParseDictionaryFailure,
+	ParseDictionaryResult,
+	ParseSchemaError,
+	ParseSchemaFailureData,
+	ParseSchemaResult,
+	ParseRecordResult,
+} from '@overture-stack/lectern-validation';
+
+import * as _processing from './processing';
+export * as rest from './rest';
+
+/**
+ * Processing functions will perform both data parsing and validation and then return an
+ * object with the parsed data and a list of any errors encountered.
+ *
+ * The available processing functions are concerned with data at different scales:
+ * - processRecord: will process a single data record using schema definition
+ * - processSchema: will process a collection of data records using a single schema definition
+ * - processDicitonary: will process multiple collecitons of data records, each vs a different
+ *     schema definition that is found in a dictionary definition.
+ */
+export const process = _processing;
+
+/**
+ * Validation functions will perform all restriction tests on data objects. The result will indicate
+ * if all tests passed or if there were some failures, and return a list of the failures that occurred.
+ */
+export const validate = {
+	validateField: _validation.validateField,
+	validateRecord: _validation.validateRecord,
+	validateSchema: _validation.validateSchema,
+	validateDictionary: _validation.validateDictionary,
+};
+
+/**
+ * Parsing functions will convert the an object with string values into a new object with all values properly typed
+ * to match the data types from a schema definition. This parsing process will convert values to numbers, booleans,
+ * and arrays, as required. String values may also be cleaned up to trim whitespace and match the casing of codeList
```

Comment:
`codeList` is a new concept for me. was this term explained anywhere I missed?

Reply:
Its just an enum of valid values, I am unclear on how this property name was originally selected.